p2p.go

```diff
@@ -380,9 +408,22 @@ func NewHost(ctx context.Context, options ...Option) (*Host, error) {
 }
 
 // JoinOverlay triggers the host to join the DHT overlay
-func (h *Host) JoinOverlay(ctx context.Context) {
+func (h *Host) JoinOverlay(ctx context.Context) error {
 	routingDiscovery := discovery.NewRoutingDiscovery(h.kad)
-	discovery.Advertise(ctx, routingDiscovery, h.kadKey.String())
+	h.peerManager = newPeerManager(h.host, routingDiscovery, h.cfg.GroupID, withMaxPeers(100), withBlacklistTolerance(3))
+	return h.peerManager.Advertise(ctx)
+}
+
+// TODO: stop goroutine when p2p stop()
+// FindPeers connect host to the peers with the same groupID
+func (h *Host) FindPeers(ctx context.Context) error {
+	if h.peerManager == nil {
+		return errNoPeerManager
+	}
```

Comment:
can remove this check and `errNoPeerManager`? `h.peerManager` will be created inside `New()`